setup.py

```diff
@@ -300,7 +300,8 @@ def check_linker_need_libatomic():
 asm_files = []
 
 asm_key = ''
-if BUILD_WITH_BORING_SSL_ASM and not BUILD_WITH_SYSTEM_OPENSSL:
+if (BUILD_WITH_BORING_SSL_ASM.upper() == 'TRUE'
+    or BUILD_WITH_BORING_SSL_ASM == '1') and not BUILD_WITH_SYSTEM_OPENSSL:
```

Comment:
nit: Technically, users can set this flag to any value and they should be evaluated into True, except `False, 0, ''`. How about we detect the falsify values intead?

Reply:
done.